### PR TITLE
Make ComputeFileRange return error on inverted byte/line ranges

### PR DESCRIPTION
**Pull request by slimsag**
This change fixes a panic where the returned byte ranges are invalid as the user passed lines or byte ranges that are in bounds (greater than zero, less than `len(data)`) but are inverted (e.g. `data[4:2]` instead of `data[2:4]`). This fixes a panic we were seeing in prod.

Also add many tests to improve catching these sort of cases in the future / avoid regressions.

### CODE REVIEW

File: vcsclient/range_test.go [hunk 1]
```diff
@@ -48,6 +48,11 @@ func TestComputeFileRange(t *testing.T) {
 			opt:  GetFileOptions{FileRange: FileRange{StartLine: 2, EndLine: 2}},
 			want: FileRange{StartLine: 2, EndLine: 2, StartByte: 2, EndByte: 4},
 		},
+		"OOB end line": {
+			data: []byte("a\nb"),
+			opt:  GetFileOptions{FileRange: FileRange{StartLine: 0, EndLine: 999999}},
+			want: FileRange{StartLine: 2, EndLine: 2, StartByte: 2, EndByte: 4},
```
Comment (beyang) [inline vcsclient/range_test.go]: `StartLine: 1, EndLine: 2, StartByte: 0, EndByte: 3`?

**Comment by beyang** [comment]
otherwise lgtm

File: vcsclient/range_test.go (hunk 1 above)
Comment (slimsag) [inline vcsclient/range_test.go]: Yep, done.

**Comment by shurcooL** [comment]
For posterity, this was merged as https://github.com/sourcegraph/vcsstore/commit/2068dbc66d6eae891d01afd69d12ab22e91b2ae0, not abandoned.

**Pull request closed without merging**